src/main.js

```diff
@@ -0,0 +1,162 @@
+const selectors = {
+    INPUT: '.keyboard-input',
+    CHAR_KEY: '.char',
+    ACTIVATABLE_KEY: '.activatable-key',
+};
+
+const classes = {
+    KEYBOARD: 'keyboard',
+    HIDDEN: 'hidden',
+    KEYBOARD_KEYS: 'keyboard-keys',
+    KEY: 'key',
+    WIDE_KEY: 'wide-key',
+    ACTIVATABLE_KEY: 'activatable-key',
+    EXTRA_WIDE_KEY: 'extra-wide-key',
+    DARK_KEY: 'dark-key',
+    CHAR_KEY: 'char',
+    ACTIVE_KEY: 'active-key',
+};
+
+const keyboardInputElement = document.querySelector(selectors.INPUT);
+let capsLock = false;
+
+const keysHandlers = {
+    backspace: () => {
+        keyboardInputElement.value = keyboardInputElement.value.slice(0,keyboardInputElement.value.length - 1);
+    },
+    keyboard_return: () => {
+        keyboardInputElement.value += '\n';
+    },
+    check_circle: () => {
+        keyboardElement.classList.add(classes.HIDDEN);
+    },
+    space_bar: () => {
+        keyboardInputElement.value += ' ';
+    },
+    keyboard_capslock: () => {
+        const buttons = document.querySelectorAll(selectors.CHAR_KEY);
+        const capsButton = document.querySelector(selectors.ACTIVATABLE_KEY);
+        
+        capsLock = !capsLock;
+        buttons.forEach(el => {
+            el.textContent = capsLock ? el.textContent.toUpperCase() : el.textContent.toLowerCase();
+        });
+        capsButton.classList.toggle(classes.ACTIVE_KEY);
+    },
+
+    default: (e) => {
+        keyboardInputElement.value += e.target.innerHTML;
+    }
+
+};
+
+let keyboardElement;
+let keyContainerElement;
+
+const createIconHTML = (icon_name) => {
+    return `<span class='material-icons'>${icon_name}</span>`;
+};
+
+function determineKeyValue(e) {    
+    let keyHandler = keysHandlers[e.target.innerText];
+
+    if (keyHandler === undefined) {
+        keyHandler = keysHandlers.default;
+    }
+    
+    if (e.target.parentNode.className !== classes.KEYBOARD) {
+        keyHandler(e);        
+    }
+}
+ 
+function listenKeyboardClick() {
+    keyboardElement.addEventListener('click', determineKeyValue);
+}
+
+function createKeys() {
+    const fragment = document.createDocumentFragment();
+    const chars = [
+        'q', 'w', 'e', 'r', 't', 'y', 'u', 'i', 'o', 'p',
+        'a', 's', 'd', 'f', 'g', 'h', 'j', 'k', 'l',
+        'z', 'x', 'c', 'v', 'b', 'n', 'm'
+    ];
+    const keyLayout = [
```

Comment:
I would change this approach and store array of objects with additional data like classes and icon data
In this case we will get rid or switch statement
Let's discuss on the call